### PR TITLE
Fix: Controlled TextInput with non-string placeholder

### PR DESCRIPTION
**Pull request by miguelmarcondesf**
Signed-off-by: miguel marcondes filho <miguelmarcondesfilho@gmail.com>

#### What does this PR do?
Fix TextInput with non-string placeholder, it was showing placeholder even after the user starts typing.

#### Where should the reviewer start?
`src/js/components/TextInput/TextInput.js`

#### What testing has been done on this PR?
`yarn storybook` and `yarn test`

#### How should this be manually tested?
`yarn storybook`

#### What are the relevant issues?
#4729 

#### Screenshots (if appropriate)
Before:
image
After fixing:
image
image

#### Do the grommet docs need to be updated?
No

#### Should this PR be mentioned in the release notes?
I don't think so

#### Is this change backwards compatible or is it a breaking change?
It is backwards compatible

### CODE REVIEW

**Review by ShimiSun — COMMENTED** [review]
LGTM! Thanks, buddy, and welcome to Grommet! 💟 
Can you please exclude the change on the `yarn.lock` file? this file shouldn't change as part of this PR.

File: yarn.lock [hunk 1]
```diff
@@ -13001,3 +13001,4 @@ yarn-or-npm@^3.0.1:
   dependencies:
     cross-spawn "^6.0.5"
     pkg-dir "^4.2.0"
+
```
Comment (ShimiSun) [inline yarn.lock:13004]: Can you please remove this file from this PR so it wouldn't have any changes?

Comment (miguelmarcondesf) [reply yarn.lock]: Hey @ShimiSun sorry for that, I think now its ok

Comment (miguelmarcondesf) [reply yarn.lock]: For some reason, I was getting an extra extra line haha

Comment (ShimiSun) [reply yarn.lock]: No worries

**Review by ShimiSun — APPROVED** [review]

**Review by ericsoderberghp — APPROVED** [review]

**Comment by ShimiSun** [comment]
Thank you for your contribution! 💟

**Pull request merged**